### ISSUE TITLE
ImportError: paddle/fluid/core_noavx.so: undefined symbol: shm_unlink

### ISSUE THREAD

**Issue by kwokyto** [PaddlePaddle/Paddle#44621]
### 问题描述 Issue Description

When running PaddleSpeech on RaspberryPi, I am given an ImportError regarding noavx in Paddle. I have followed instructions from [QEngineering](https://github.com/Qengineering/Paddle-Raspberry-Pi) to download the paddlepaddle wheel for aarch64 machines.

## Reproduction Steps
```bash
## Installation
$ sudo apt-get install cmake wget libatlas-base-dev libopenblas-dev libblas-dev liblapack-dev patchelf gfortran
$ pip install Cython six requests wheel pyyaml protobuf==3.13.0
$ pip install -U setuptools
$ wget https://github.com/Qengineering/Paddle-Raspberry-Pi/raw/main/paddlepaddle-2.3.1-cp39-cp39-linux_aarch64.whl
$ pip install paddlepaddle-2.3.1-cp39-cp39-linux_aarch64.whl
$ pip install pytest-runner -i https://pypi.tuna.tsinghua.edu.cn/simple
$ pip install paddlespeech -i https://pypi.tuna.tsinghua.edu.cn/simple

## Execution
$ paddlespeech asr –model transformer_librispeech –lang en –input sample.wav -v
Error: Can not import noavx core while this file exists: /home/elock/archiconda3/envs/paddle/lib/python3.9/site-packages/paddle/fluid/core_noavx.so
Traceback (most recent call last):
  File "/home/elock/archiconda3/envs/paddle/bin/paddlespeech", line 5, in <module>
    from paddlespeech.cli.entry import _execute
  File "/home/elock/archiconda3/envs/paddle/lib/python3.9/site-packages/paddlespeech/cli/__init__.py", line 16, in <module>
    from .asr import ASRExecutor
  File "/home/elock/archiconda3/envs/paddle/lib/python3.9/site-packages/paddlespeech/cli/asr/__init__.py", line 14, in <module>
    from .infer import ASRExecutor
  File "/home/elock/archiconda3/envs/paddle/lib/python3.9/site-packages/paddlespeech/cli/asr/infer.py", line 24, in <module>
    import paddle
  File "/home/elock/archiconda3/envs/paddle/lib/python3.9/site-packages/paddle/__init__.py", line 25, in <module>
    from .framework import monkey_patch_variable
  File "/home/elock/archiconda3/envs/paddle/lib/python3.9/site-packages/paddle/framework/__init__.py", line 17, in <module>
    from . import random  # noqa: F401
  File "/home/elock/archiconda3/envs/paddle/lib/python3.9/site-packages/paddle/framework/random.py", line 16, in <module>
    import paddle.fluid as fluid
  File "/home/elock/archiconda3/envs/paddle/lib/python3.9/site-packages/paddle/fluid/__init__.py", line 36, in <module>
    from . import framework
  File "/home/elock/archiconda3/envs/paddle/lib/python3.9/site-packages/paddle/fluid/framework.py", line 37, in <module>
    from . import core
  File "/home/elock/archiconda3/envs/paddle/lib/python3.9/site-packages/paddle/fluid/core.py", line 366, in <module>
    raise e
  File "/home/elock/archiconda3/envs/paddle/lib/python3.9/site-packages/paddle/fluid/core.py", line 314, in <module>
    from . import core_noavx
ImportError: /home/elock/archiconda3/envs/paddle/lib/python3.9/site-packages/paddle/fluid/core_noavx.so: undefined symbol: shm_unlink
```

## OS
```
$ uname -a
Linux raspberrypi 5.15.32-v8+ #1538 SMP PREEMPT Thu Mar 31 19:40:39 BST 2022 aarch64 GNU/Linux
```

## GPU
None

## Environment
- Conda 4.5.12
- Python 3.9.13
- GCC 10.2.1
- PaddlePaddle 2.3.1
- PaddleSpeech 1.0.1

## Misc
```
paddle-bfloat      0.1.7
paddle2onnx        0.9.2
paddleaudio        1.0.1
paddlefsl          1.1.0
paddlenlp          2.3.4
paddlepaddle       2.3.1
paddlespeech       1.0.1
paddlespeech-feat  0.1.0
```

**Comment by thisjiang** [comment]
Hi! Can you run `python -c 'import paddle;paddle.utils.run_check()'`?

**Comment by kwokyto** [comment]
> Hi! Can you run `python -c 'import paddle;paddle.utils.run_check()'`?

it seems like i still get the same error
```
$ python -c 'import paddle;paddle.utils.run_check()'
Error: Can not import noavx core while this file exists: /home/elock/archiconda3/envs/paddle/lib/python3.9/site-packages/paddle/fluid/core_noavx.so
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/elock/archiconda3/envs/paddle/lib/python3.9/site-packages/paddle/__init__.py", line 25, in <module>
    from .framework import monkey_patch_variable
  File "/home/elock/archiconda3/envs/paddle/lib/python3.9/site-packages/paddle/framework/__init__.py", line 17, in <module>
    from . import random  # noqa: F401
  File "/home/elock/archiconda3/envs/paddle/lib/python3.9/site-packages/paddle/framework/random.py", line 16, in <module>
    import paddle.fluid as fluid
  File "/home/elock/archiconda3/envs/paddle/lib/python3.9/site-packages/paddle/fluid/__init__.py", line 36, in <module>
    from . import framework
  File "/home/elock/archiconda3/envs/paddle/lib/python3.9/site-packages/paddle/fluid/framework.py", line 37, in <module>
    from . import core
  File "/home/elock/archiconda3/envs/paddle/lib/python3.9/site-packages/paddle/fluid/core.py", line 366, in <module>
    raise e
  File "/home/elock/archiconda3/envs/paddle/lib/python3.9/site-packages/paddle/fluid/core.py", line 314, in <module>
    from . import core_noavx
ImportError: /home/elock/archiconda3/envs/paddle/lib/python3.9/site-packages/paddle/fluid/core_noavx.so: undefined symbol: shm_unlink
```

**Comment by thisjiang** [comment]
Is this inference or training?

**Comment by kwokyto** [comment]
> Is this inference or training?

This is inference, just want to run a wav file through it

**Comment by thisjiang** [comment]
This seems your environment is not the same as the whl supporter's, you can run `cat /proc/cpuinfo | grep -i avx` to check whether your cpu support avx or not. Just to be sure, I suggest you should compile the paddle from the source code.

**Comment by thisjiang** [comment]
Hi！If only for inference, we suggest you using [Paddle-Lite](https://github.com/PaddlePaddle/Paddle-Lite), Paddle Lite confirmed easy to perform inference on RaspberryPi.

**Comment by sfraczek** [comment]
I have the same issue with newest PaddlePaddle except that is is not core_noavx but core_avx. When I used older version (from 13 July) it works fine. 
Running
`cat /proc/cpuinfo | grep -i avx  `
i have avx avx2 avx512f avx512dq avx512cd avx512bw avx512vl

I'm running some infer.py from PaddleNLP

**Comment by thisjiang** [comment]
@sfraczek PaddlePaddle support avx(default) and noavx whl. If your cpu support avx, you can pip install paddlepaddle by default, otherwise you should pip install with `-f https://www.paddlepaddle.org.cn/whl/linux/mkl/noavx/stable.html --no-index --no-deps
`. Ref to https://www.paddlepaddle.org.cn/documentation/docs/zh/install/pip/linux-pip.html. Thanks.

**Comment by sfraczek** [comment]
As a contributor I'm trying to build from source. On Ubuntu 20

**Comment by thisjiang** [comment]
@sfraczek As [Compile Option Table](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/install/Tables.html#Compile), you can using `WITH_AVX=ON` flag in cmake command to build paddle whl with avx . And you can find all cmake flag in [configure.cmake](https://github.com/PaddlePaddle/Paddle/blob/develop/cmake/configure.cmake).

**Comment by sfraczek** [comment]
Changing flag didn't help. I tried in docker instead of my ubuntu 20 pc and now it worked
paddlepaddle/paddle:latest-dev-cuda10.1-cudnn7-gcc82 (ubuntu 16). I wonder why.

**Comment by sfraczek** [comment]
I've heard that the problem is fixed by using gcc version <=8.2

**Comment by paulinagacek** [comment]
In my case problem occurred with gcc 8.4 and above as well as with gcc 7.5. It was solved by using gcc 8.2

**Comment by Vector-Cross** [comment]
same problem when test as below
```
python3 -c "import paddle; paddle.utils.run_check()"
Error: Can not import paddle core while this file exists: /opt/conda/envs/pytorch/lib/python3.7/site-packages/paddle/fluid/libpaddle.so
ImportError: /opt/conda/envs/pytorch/lib/python3.7/site-packages/paddle/fluid/libpaddle.so: undefined symbol: shm_unlink
NameError: name 'libpaddle' is not defined
```
My gcc version is 9.4.
I will change to gcc 8.2 build again.
